lib/jekyll_pages_api/generator.rb

```diff
@@ -10,7 +12,12 @@ def initialize(site)
     end
 
     def pages
-      self.site.pages.map{|page| Page.new(page) }.select(&:html?)
+      result = Array.new
+      self.site.each_site_file do |p|
```

Comment:
Is this an internal Jekyll API? Seems like a good find, though it could be more fragile. We should keep an eye out for it changing.